source/common/secret/secret_manager_impl.cc

```diff
@@ -79,5 +84,57 @@ SecretManagerImpl::findOrCreateCertificateValidationContextProvider(
                                                     secret_provider_context);
 }
 
+// TODO: question, what's the handling of static inlined stuff? they're just internal constructs
+// not needed to be exposed, maybe?
+ProtobufTypes::MessagePtr SecretManagerImpl::dumpSecretConfigs() {
+  auto config_dump = std::make_unique<envoy::admin::v2alpha::SecretsConfigDump>();
+  auto providers = certificate_providers_.allSecretProviders();
+  for (const auto& cert_secrets : providers) {
+    const auto& secret_data = cert_secrets->secretData();
+    const auto& tls_cert = cert_secrets->secret();
+    const auto& dynamic_secret = config_dump->mutable_dynamic_secrets()->Add();
+    const auto& secret = dynamic_secret->mutable_secret();
```

Comment:
the latter two `const auto&` resolves to `const Secret*&` which doesn't makes sense. use `auto`

Reply:
Done.